### DISCUSSION TITLE
Which files can be patched by image automation?

### DISCUSSION THREAD

**Discussion by derrickburns** [fluxcd/flux2#1826] [Q&A] [answered]
I understand how to provide a path within a GitRepository to indicate to the Image Automation controller the directory in which to patch files.

I understand that only files that look like Kubernetes resources get patched.  I *presume* that this means files ending in `.yaml` that contain `apiVersion` and `kind` keys.  Are there other constraints?  

Can I patch files within a helm chart?  I presume that I cannot patch the helm template files themselves because they are templates, not resources.  However, if I have a subdirectory in the chart that has files that are valid K8s resources, will the Image Automation controller patch them?

**Comment by stefanprodan** [comment, marked as answer]
If those YAMLs are valid Kubernetes objects (no helm templating), then yes Flux can patch them.

**Comment by squaremo** [comment]
I'm going to turn this into a discussion, to surface it as a Q&A
